dvc/ui/__init__.py

```diff
@@ -69,12 +69,12 @@ def write(
         self,
         *objects: Any,
         style: str = None,
-        sep: str = SEP,
-        end: str = NEWLINE,
+        sep: str = None,
+        end: str = None,
         file: TextIO = None,
         flush: bool = False,
     ) -> None:
-        if self.disabled:
+        if not self._enabled:
```

Comment:
Q: How important is thread-safety for us while printing? `logger` is thread safe, but this is not.

Reply:
So, ultimately, we are intending to use `ui.write` everywhere we use `logger` now? If so, I would say thread safety is pretty important as we already have some parts of code that utilize `ThreadPoolExecutor` and log something down the line (for example, [`_upload_plans`](https://github.com/iterative/dvc/blob/f2ba67aece1d5ff3bc5aaf2d40948606c3f7e5c8/dvc/remote/base.py#L370).

Reply:
thanks, @pared. Will come to it as we start using it in multithreaded applications.